ananse/influence.py

```diff
@@ -259,40 +376,91 @@ def plot_influscore(infile, outfile):
             logger.warning("You only provide the target network!")
-        elif Gaf is None and Gbf is not None:
-            self.G = read_network(Gbf, edges=edges)
+        elif GRN_target_file is None and GRN_source_file is not None:
+            self.G = read_network(GRN_source_file, edges=edges)
             logger.warning("You only provided the source network!")
-        elif Gaf is None and Gbf is None:
+        elif GRN_target_file is None and GRN_source_file is None:
             logger.warning("You should provide at least one ANANSE network file!")
         else:
-            G1 = read_network(Gbf, edges=edges)
-            G2 = read_network(Gaf, edges=edges)
-            self.G = difference(G2, G1)
+            logger.info(
+                f"Reading network(s), using the union of the top {edges} edges of each GRN."
+            )
+            top_int_source = read_top_interactions(
+                GRN_source_file, edges=edges, GRNsort_column=GRNsort_column
+            )
+            top_int_target = read_top_interactions(
+                GRN_target_file, edges=edges, GRNsort_column=GRNsort_column
+            )
+            top_int = set.union(top_int_source, top_int_target)
+            G1_source = read_network(
+                GRN_source_file,
+                interactions=top_int,
+                GRNsort_column=GRNsort_column,
+                full_output=full_output,
+            )
+            G2_target = read_network(
+                GRN_target_file,
+                interactions=top_int,
+                GRNsort_column=GRNsort_column,
+                full_output=full_output,
+            )
+            self.G = difference(G1_source, G2_target, full_output=full_output)
             logger.info(f"Differential network has {len(self.G.edges)} edges.")
-
         # Load expression file
         self.expression_change = read_expression(degenes)
-
         self.outfile = outfile
-
         # Filter TFs
         self.filter = filter
 
-    def save_reg_network(self, filename):
+    def save_reg_network(self, filename, full_output=False):
         """Save the network difference between two cell types to a file."""
-
         with open(filename, "w") as nw:
-            for (u, v, d) in self.G.edges(data=True):
-                nw.write(u + "\t" + v + "\t" + str(d["weight"]) + "\n")
+            if full_output == False:
+                nw.write(f"TF\ttarget\tweight\n")
+                for (u, v, d) in self.G.edges(data=True):
+                    nw.write(u + "\t" + v + "\t" + str(d["weight"]) + "\n")
+            if full_output:
+                logger.info("output full diff network")
+                nw.write(
+                    f"tf\ttarget\tweight\tweight_source\tweight_target\ttf_expr_source\ttf_expr_target\ttg_expr_source\ttg_expr_target\twb_source\twb_target\tsource_tf_act\ttarget_tf_act\n"
+                )
+                for (u, v, d) in self.G.edges(data=True):
```

Comment:
In the future, we should check if we can save the network without looping. Maybe with a [builtin write function](https://networkx.org/documentation/stable/reference/readwrite/edgelist.html).

For now, this looks pretty clear to me!